### PR TITLE
Add test cases for Ansible Tower job retireable? method.

### PR DESCRIPTION
**Pull request by lfu**
Depends on https://github.com/ManageIQ/manageiq/pull/18329.

https://bugzilla.redhat.com/show_bug.cgi?id=1650437.

@miq-bot add_label bug, blocker, hammer/yes

### CODE REVIEW

**Comment by d-m-u** [comment]
Thanks, @lfu!

File: spec/support/ansible_shared/automation_manager/job.rb [hunk 1]
```diff
@@ -59,6 +59,7 @@
           expect(job.job_template).to          eq(template)
```
Comment (gmcculloug) [inline spec/support/ansible_shared/automation_manager/job.rb]: @lfu Please review test failures.  Also, this file should be named `job_spec.rb`

cc @tinaafitz

Comment (lfu) [reply spec/support/ansible_shared/automation_manager/job.rb]: @gmcculloug Looks like files containing shared examples are named as regular ruby files. 
Check this [in core repo](https://github.com/ManageIQ/manageiq/tree/924a89012a1a68d0cda31e19e14cd49188d53a04/spec/support/examples_group) and [in Ansible Tower repo](https://github.com/ManageIQ/manageiq-providers-ansible_tower/tree/f9bdaeab214e2b9775d8ab736f4035307f0886f9/spec/support/ansible_shared).

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 77dba50b20b83e0949c88026ca05b593e6ba9e81
Author: Greg McCullough <gmccullo@redhat.com>
Date:   Tue Jan 8 13:01:10 2019 -0500

    Merge pull request #150 from lfu/adding_ci_feature_mixin_to_job_1650437
    
    Add test cases for Ansible Tower job retireable? method.
    
    (cherry picked from commit 2e3e2d0b04c3f38cb109b759128654afb8f09a8b)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1650437
```

**Pull request merged**